client/src/components/Donation/donation.css

```diff
@@ -305,6 +305,18 @@ li.disabled > a {
   justify-content: space-between;
 }
 
+@media (max-width: 500px) {
+  .edit-amount-confirmation {
+    width: 100% !important;
+    flex-direction: column;
+    align-items: center;
+  }
+
+  .close-button {
+    margin-top: 10px;
+  }
+}
```

Comment:
Changing the styles based on the width does not work since the localized versions might have larger translations.
If you want to continue with flexbox, I encourage you to experiment with flex-wrap.